crates/polars-plan/src/plans/options.rs

```diff
@@ -217,10 +209,12 @@ impl FunctionOptions {
     }
 
     pub fn is_elementwise(&self) -> bool {
-        self.collect_groups == ApplyOptions::ElementWise
-            && !self
-                .flags
-                .contains(FunctionFlags::CHANGES_LENGTH | FunctionFlags::RETURNS_SCALAR)
+        matches!(
+            self.collect_groups,
+            ApplyOptions::ElementWise | ApplyOptions::ApplyList
```

Comment:
I don't fully understand this but we apparently consider `ApplyList` as elementwise according to the test suite

Reply:
I think we should see when we hit this. I believe only in a few python udf cases. I think we would never hit the streaming engine with this.

Reply:
Yes, this is hit by `map_elements()` / `map_batches(.., agg_list=True)`. This does hit the streaming engines because `FunctionOptions::is_elementwise()` gets used by all of the engines during physical plan creation / IR lowering (indirectly through `fn is_elementwise(_rec)`).

I don't think `ApplyList` is strictly elementwise, but both the in-memory and existing streaming engine currently expect it to be identified as such, so I've made it the default here. For the new-streaming engine I added an override below.